services/util/client.js

```diff
@@ -7,25 +8,42 @@ const defaultParams = { action: 'query', format: 'json' };
 const apiPath = 'w/api.php';
 
 function transform(data) {
-  const { query } = data;
-  return query;
+  assert.ok(data.query, errors.emptyResponse);
+  const {
+    query: { pages },
+  } = data;
+  assert.ok(pages, errors.emptyResponse);
+  return pages;
 }
 
 function handleError(error) {
   if (!error.response) throw new Error(errors.apiUnavailabe);
   throw error;
 }
 
-async function queryApi({ client, wikiUrl, params }) {
-  const apiUrl = Url.resolve(wikiUrl, apiPath);
+function dataIsComplete(data) {
+  return 'batchcomplete' in data;
+}
+
+async function performRequest({ client, apiUrl, params }) {
   try {
-    const { data } = await client.get(apiUrl, { params });
-    return transform(data);
+    const response = await client.get(apiUrl, { params });
+    return response;
   } catch (error) {
     return handleError(error);
   }
 }
 
+async function queryApi({ client, wikiUrl, params }, existingPages = {}) {
+  const apiUrl = Url.resolve(wikiUrl, apiPath);
+  const { data } = await performRequest({ client, apiUrl, params });
+  const pages = { ...existingPages, ...transform(data) };
```

Comment:
Hm, looking at this again, I fear this is actually a quite dump idea 🙈 

When doing an `'image'` query and the results are distributed over multiple response, the `page` will of course always be the same... The structure looks like this:

First request:
```json
{
  "continue": {
    "continue": "||",
    "imcontinue": "77118|Apple_Lisa.jpg"
  },
  "query": {
    "normalized": [
      {
        "from": "Apple_Lisa",
        "to": "Apple Lisa"
      }
    ],
    "pages": {
      "77118": {
        "images": [
          {
            "ns": 6,
            "title": "File:Ambox important.svg"
          },
          {
            "ns": 6,
            "title": "File:Apple Computer, 1983 (Lisa).jpg"
          }
        ],
        "ns": 0,
        "pageid": 77118,
        "title": "Apple Lisa"
      }
    }
  }
}
```

And the second one looks like this:
```json
{
  "continue": {
    "continue": "||",
    "imcontinue": "77118|Apple_Lisa_IO_board.JPG"
  },
  "query": {
    "normalized": [
      {
        "from": "Apple_Lisa",
        "to": "Apple Lisa"
      }
    ],
    "pages": {
      "77118": {
        "images": [
          {
            "ns": 6,
            "title": "File:Apple Lisa.jpg"
          },
          {
            "ns": 6,
            "title": "File:Apple Lisa2-IMG 1517.jpg"
          }
        ],
        "ns": 0,
        "pageid": 77118,
        "title": "Apple Lisa"
      }
    }
  }
}
```